### ISSUE TITLE
Implement defence evaluations by Tramèr et al. (2020)

### ISSUE THREAD

**Issue by beat-buesser** [Trusted-AI/adversarial-robustness-toolbox#314]
Implement defence evaluations of Tramèr et al. (2020) as notebooks using ART and identify gaps in ART.

Paper link: https://arxiv.org/abs/2002.08347

Create a separate notebook for each evaluation in a new folder in notebooks.
Create separate issues for each evaluation.

- [ ] 5. k-Winner Take All
- [ ] 6. The Odds are Odd
- [ ] 7. Are Generative Classifiers More Robust?
- [ ] 8. Robust Sparse Fourier Transform
- [ ] 9. Rethinking Softmax Cross Entropy
- [ ] 10. Error Correcting Codes
- [ ] 11. Ensemble Diversity
- [x] 12. EMPIR (implemented in issue #319)
- [ ] 13. Temporal Dependency
- [ ] 14. Mixup Inference
- [ ] 15 ME-Net
- [ ] 16. Asymmetrical Adversarial Training
- [ ] 17. Turning a Weakness into a Strength
